api/br/v1alpha1/lightweightbrgc_types.go

```diff
@@ -0,0 +1,108 @@
+// +kubebuilder:resource:categories=br
+// +kubebuilder:resource:shortName="lwbrgc"
+// +kubebuilder:selectablefield:JSONPath=`.spec.cluster.name`
+// +kubebuilder:printcolumn:name="Cluster",type=string,JSONPath=`.spec.cluster.name`
+// +kubebuilder:printcolumn:name="Strategy",type=string,JSONPath=`.spec.gcStrategy.type`
+// +kubebuilder:printcolumn:name="Synced",type=string,JSONPath=`.status.conditions[?(@.type=="Synced")].status`
+// +kubebuilder:printcolumn:name="Ready",type=string,JSONPath=`.status.conditions[?(@.type=="Ready")].status`
+// +kubebuilder:printcolumn:name="Age",type="date",JSONPath=".metadata.creationTimestamp"
+
+// LightweightBackupGC is the Schema for the lightweightbackupgc API
+type LightweightBackupGC struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   LightweightBackupGCSpec   `json:"spec,omitempty"`
+	Status LightweightBackupGCStatus `json:"status,omitempty"`
+}
+
+// LightweightBackupGCSpec defines the desired state of LightweightBackupGC.
+type LightweightBackupGCSpec struct {
+	// Cluster is a reference of tidb cluster
+	Cluster    v1alpha1.ClusterReference   `json:"cluster"`
+	GCStrategy LightweightBackupGCStrategy `json:"gcStrategy,omitempty"`
+	// Image is image of br gc, default is pingcap/tikv
+	Image       *string                         `json:"image,omitempty"`
+	PodsOverlay []LightweightBackupGCPodOverlay `json:"podsOverlay,omitempty"`
+}
+
+// LightweightBackupGCStatus defines the observed state of LightweightBackupGC.
+type LightweightBackupGCStatus struct {
+	v1alpha1.CommonStatus `json:",inline"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+
+// LightweightBackupGCList contains a list of LightweightBackupGC.
+type LightweightBackupGCList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []LightweightBackupGC `json:"items"`
+}
+
+type LightweightBackupGCStrategyType string
+
+const (
+	LightweightBackupGCStrategyTypeTieredStorage LightweightBackupGCStrategyType = "tiered-storage"
+)
+
+type LightweightBackupGCStrategy struct {
+	// +kubebuilder:validation:Enum=tiered-storage
+	Type             LightweightBackupGCStrategyType `json:"type"`
+	TieredStrategies []TieredStorageStrategy         `json:"tieredStrategies,omitempty"`
+}
+type TieredStorageStrategyName string
+
+const (
+	TieredStorageStrategyNameToT2Storage TieredStorageStrategyName = "to-t2-storage"
+	TieredStorageStrategyNameToT3Storage TieredStorageStrategyName = "to-t3-storage"
+)
+
+type TieredStorageStrategy struct {
+	// +kubebuilder:validation:Enum=to-t2-storage;to-t3-storage
+	Name TieredStorageStrategyName `json:"name"`
+	// +kubebuilder:validation:Pattern=`^\d+d$`
+	// +kubebuilder:validation:Rule="int(self[:-1]) > 0"
+	// +kubebuilder:validation:Message="Duration must be a positive number of days like '1d', '7d'"
+	TimeThreshold string `json:"timeThreshold"`
```

Comment:
If it indicate days, why not use `int` directly

Reply:
you are right

Reply:
fixed